apps/prairielearn/python/zygote.py

```diff
@@ -84,6 +87,33 @@
 prairielearn.get_unit_registry()
 
 
+# We want to conditionally allow/block importing specific modules, namely `rpy2`.
+# This custom importer will allow us to do so, and throw a custom error message.
+#
+# While this won't prevent anything more complex than an `import` statement, it
+# will make it clear to the user that they're not allowed to use `rpy2`. If they
+# try to bypass the block, it's up to them to deal with the consequences.
+class ForbidModuleMetaPathFinder(MetaPathFinder):
+    def __init__(self):
+        self.forbidden_modules: set[str] = set()
+
+    def forbid_module(self, forbidden_module: str):
+        self.forbidden_modules.add(forbidden_module)
```

Comment:
```suggestion
        self.forbidden_modules.update(forbidden_modules)
```
Nit: The primary use case is to forbid modules coming from a list (or another iterable), so having a function `forbid_modules` that takes this in would be nice.